### DISCUSSION TITLE
Implement a Generic Resource Editing interface

### DISCUSSION THREAD

**Discussion by f-odhiambo** [opensrp/fhircore#1464] [Ideas]
Related to ticket https://github.com/opensrp/fhircore/issues/1436, we had a discussion on the ticket on Edit Family and had the following thoughts, we can complete the edit functionality in a configurable way such that the functionality can be reused e.g. use it for Edit Patient details too, that way it makes the app more configurable. Thus

1. Fix how to handle questionnaires
2. on Create/Edit
3. Then we can work on an Edit QR module that can do population i.e. Pre-populate from QuestionnaireResponse
4. QR should be stored alongside extracted resources, check on the current app and make sure this is enforced
5. Then edit and update the related resource
6. On the UI, we can also check who we load Questionnaire Title/ name  by fetching Questionnaire title directly from the Questionnaire during data collection but modify the title during Edit
7. Add ellipses when  we are over a certain number of characters
8. Add MLS support on this too

More ideas are welcome

**Comment by FikriMilano** [comment]
For point 2 whether a Questionnaire should create or edit response, I think we can leverage the forms configuration
https://github.com/opensrp/fhircore/tree/main/docs/configurability#structure-of-questionnaireconfig

**Comment by FikriMilano** [comment]
Could you explain how the QR workflow works? on real/CHW scenario.

**Comment by FikriMilano** [comment]
We already implement point 6 for appending edit tile.

For point 7 ellipses title, it requires code as well.

**Comment by FikriMilano** [comment]
For MLS support, are you implying the whole Questionnaire?

**Reply by f-odhiambo** [reply]
Yes when you switch language all questionnaire artifacts are translated including the title and buttons
